backend/recommendation/user_recommendation.js

```diff
@@ -16,18 +17,17 @@ const calculateAvailabilityScore = (
 
 const calculateArrayScore = (userValue, groupValue) => {
   let matchedValue = [];
-  const flatUser = userValue.map((value) =>
-    Object.values(value).filter((val) => typeof val === "string")
-  );
-  const flatGroup = groupValue.map((value) =>
-    Object.values(value).filter((val) => typeof val === "string")
-  );
-
-  for (let i = 0; i < flatGroup.length; i++) {
-    matchedValue = matchedValue.concat(
-      flatUser.filter((value) => flatGroup[i][0].includes(value))
+  const flatArray = (arrayOfObjects) => {
+    return arrayOfObjects.flatMap((value) =>
+      Object.values(value).filter((val) => typeof val === "string")
     );
-  }
+  };
+  const flatUser = flatArray(userValue);
+  const flatGroup = flatArray(groupValue);
+
+  matchedValue = matchedValue.concat(
```

Comment:
You should be able to remove this concatenation and just set `matchedValue` to be the result of the filtering

Reply:
I have updated it here: https://github.com/izuchukwumba/peered/pull/16/commits/5e0535ebc6030d0598a05665d1223db346de2faa